parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueWriters.java

```diff
@@ -165,9 +178,65 @@ public void writeDouble(int repetitionLevel, double value) {
     }
   }
 
+  private static class FloatWriter extends UnboxedWriter<Float> {
+    private final ColumnDescriptor desc;
+    private long nanCount;
+
+    private FloatWriter(ColumnDescriptor desc, Type.ID id) {
+      super(desc, id);
+      this.desc = desc;
+      this.nanCount = 0;
+    }
+
+    @Override
+    public void write(int repetitionLevel, Float value) {
+      super.write(repetitionLevel, value);
+      if (Float.compare(Float.NaN, value) == 0) {
```

Comment:
I would prefer to use `float.isNaN()` instead because it is more direct. The implementation of compare does a few comparisons because it needs to work for normal values, but `Float.isNaN` just returns `value != value`.

Reply:
Didn't notice that this method exists, thank you!